src/main/java/com/uid2/admin/vertx/service/ClientKeyService.java

```diff
@@ -208,6 +191,46 @@ private void handleClientListBySite(RoutingContext rc) {
         }
     }
 
+    private void handleClientByKeyId(RoutingContext rc) {
+        try {
+            String keyId;
+
+            try{
+                keyId = rc.queryParam("keyId").get(0);
+            } catch (Exception e) {
+                ResponseUtil.error(rc, 400, "unable to parse keyId " + e.getMessage());
+                return;
+            }
+
+            LegacyClientKey clientKey = this.clientKeyProvider.getAll().stream().filter(legacyClientKey -> legacyClientKey.getKeyId().equals(keyId)).findFirst().orElse(null);
+
+            if (clientKey == null){
+                ResponseUtil.error(rc, 404, "unable to find key");
+                return;
+            }
+
+            rc.response()
+                    .putHeader(HttpHeaders.CONTENT_TYPE, "application/json")
+                    .end(getClientReturnObject(clientKey).encode());
+        } catch (Exception e) {
+            rc.fail(500, e);
+        }
+    }
+
+    private static JsonObject getClientReturnObject(LegacyClientKey clientKey) {
```

Comment:
when we did this for `SiteService.java`, we called it `createSiteJsonObject`. Perhaps we could call this one `createClientKeyJsonObject` or similar?